File: abc/abc333/d.cpp
#include <cassert>
#include <cfenv>
#include <cfloat>
#include <ciso646>
#include <clocale>
#include <csetjmp>
#include <csignal>
#include <cstdbool>
#include <cinttypes>
#include <charconv>
#include <typeindex>
#include <any>
#include <scoped_allocator>
#include <forward_list>
#include <list>
#include <map>
#include <set>
#include <valarray>
#include <variant>
#include <unordered_map>
#include <unordered_set>
#include <queue>
#include <condition_variable>
#include <shared_mutex>
#include <codecvt>
#include <future>
#include <regex>
#include <iostream>
#include <random>
#include <ctgmath>
#include <fstream>
using namespace std;

int count_tree_size(int current_node, vector< vector<int> > &edges, vector<bool> &visited) {
    queue<int> nodes_in_tree;
    int size = 1;
    visited.at(current_node) = true;
    nodes_in_tree.push(current_node);
    while (not nodes_in_tree.empty()) {
        int current_node = nodes_in_tree.front();
        nodes_in_tree.pop();
        for (int next_node : edges.at(current_node)) {
            if (visited.at(next_node)) {
                continue;
            }
            nodes_in_tree.push(next_node);
            visited.at(next_node) = true;
            size++;
        }
    }
    return size;
}

int main() {
    int n, u, v;
    cin >> n;
    vector< vector<int> > edges(n);
    for (int i = 0; i < n - 1; i++) {
        cin >> u >> v;
        u--;
        v--;
        edges[u].push_back(v);
        edges[v].push_back(u);
    }

    vector<int> tree_size_list;
    for (int next_node : edges.at(0)) {
        vector<bool> visited(n, false);
        visited.at(0) = true;
        int current_size = count_tree_size(next_node, edges, visited);
        tree_size_list.push_back(current_size);
    }
    sort(tree_size_list.begin(), tree_size_list.end());
    int answer = 1;
    for (int i = 0; i < tree_size_list.size() - 1; i++) {
        answer += tree_size_list.at(i);
    }
    cout << answer << endl;
}

File: abc/abc333/d.py
def count_tree_size(node, edges, visited):
    nodes_in_tree = [node]
    size = 1
    visited[node] = True
    while nodes_in_tree:
        current_node = nodes_in_tree.pop()
        for next_node in edges[current_node]:
            if visited[next_node]:
                continue
            nodes_in_tree.append(next_node)
            visited[next_node] = True
            size += 1
    return size


def main():
    n = int(input())
    edges = [[] for _ in range(n)]
    for _ in range(n - 1):
        u, v = map(int, input().split())
        edges[u - 1].append(v - 1)
        edges[v - 1].append(u - 1)

    tree_size_list = []
    for next_node in edges[0]:
        visited = [False] * n
        visited[0] = True
        current_size = count_tree_size(next_node, edges, visited)
        tree_size_list.append(current_size)
    tree_size_list = sorted(tree_size_list)
    print(sum(tree_size_list[:-1]) + 1)


if __name__ == "__main__":
    main()
